### ISSUE TITLE
Version 0.15 and higher doesn't work with pydantic 1

### ISSUE THREAD

**Issue by asherchenkov** [15r10nk/inline-snapshot#158]
[Here](https://github.com/15r10nk/inline-snapshot/commit/f6a5ca6db5037b7b12d774d175d47f2d4d8a1365#diff-3d534b4b483da56ebd4e6724e60f2395ee2289b81bd3391ff9ad2a2be6397ab7R260) pydnatic core is imported. I causes exception if you use pyndatic 1

**Comment by 15r10nk** [comment]
The problem is that pydantic 1 & 2 produced different snapshots before 0.15.

example:
``` python
from pydantic import BaseModel

from inline_snapshot import snapshot

class M(BaseModel):
    a:int
    pass


def test():
    assert M(a=5)==snapshot()
```

#### uvx --with "pydantic>=2.1.0" --with inline-snapshot==0.14.0 pytest pydantic_v1.py --inline-snapshot=report
```
...
╭─────────────────────────────────── pydantic_12.py ───────────────────────────────────╮
│ @@ -9,5 +9,5 @@                                                                      │
│                                                                                      │
│                                                                                      │
│                                                                                      │
│  def test():                                                                         │
│ -    assert M(a=5)==snapshot()                                                       │
│ +    assert M(a=5)==snapshot(M(a=5))                                                 │
╰──────────────────────────────────────────────────────────────────────────────────────╯
```


#### uvx --with pydantic==1.10.19 --with inline-snapshot==0.13.0 pytest pydantic_v1.py --inline-snapshot=report

```
...
╭─────────────────────────────────── pydantic_12.py ───────────────────────────────────╮
│ @@ -9,5 +9,5 @@                                                                      │
│                                                                                      │
│                                                                                      │
│                                                                                      │
│  def test():                                                                         │
│ -    assert M(a=5)==snapshot()                                                       │
│ +    assert M(a=5)==snapshot({"a": 5})                                               │
╰──────────────────────────────────────────────────────────────────────────────────────╯
```

The reason for this difference is described in issue #137 


I would prefer the new snapshots which I get with pydantic v1. What do you think?

I closed #137 because I found no way to make it work. But I can take a look again. Maybe I have another Idea.

#### another problem is this:

``` python
from pydantic import BaseModel

from inline_snapshot import snapshot
from dirty_equals import IsInt

class M(BaseModel):
    a:int
    pass


def test():
    assert M(a=5)=={"a":IsInt()} # works for pydantic v1 but not v2 (can not compare with dict)
    assert M(a=5)==M(a=IsInt())  # does not work at all (validation error)
```

I would like to support both version in the same way but it looks difficult to do.

I'm not a pydantic expert. Please tell me if you have an idea how this should be handled.

**Comment by 15r10nk** [comment]
solved in 0.18.0

I was also able to implement the same snapshot format for both pydantic versions.
